.github/workflows/reusable_workflow.yaml

```diff
@@ -58,21 +58,23 @@ jobs:
 
       - name: Determine diff
         run: |
-          set +e
-          diff README.md README_TMP.md > arcaflow-docsgen.diff
-          set -e
-          echo "README_DIFF=$(cat arcaflow-docsgen.diff | wc -l)" >> $GITHUB_ENV
+          echo "README_DIFF=$(cmp -s README.md README_TMP.md ; echo $?)" >> $GITHUB_ENV
       
-      - name: Update README.md if necessary
-        if: env.README_DIFF != 0 && !github.ref_protected
+      - name: Update README.md if appropriate
+        # The github.head_ref variable is only available for pull requests,
+        # however, we probably shouldn't be updating the branch for anything
+        # which isn't a PR, anyway!
+        if: env.README_DIFF == 1 && github.event_name == 'pull_request'
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
           GH_EMAIL: arcalot@redhat.com
           GH_USERNAME: arcabot
         run: |
-          mv README_TMP.md README.md
           git config --global user.name $GH_USERNAME
           git config --global user.email $GH_EMAIL
-          git add README.md
-          git commit -m 'Automatic update of README.md by arcaflow-docsgen arcabot'
-          git push
+          git fetch origin ${{ github.head_ref }}
+          git checkout ${{ github.head_ref }}
+          mv README_TMP.md README.md
+          git commit -m 'Automatic update of README.md by arcaflow-docsgen arcabot' README.md
+          git push origin HEAD:${{ github.head_ref }}
+          git switch ${{ github.ref }}
```

Comment:
Remind me -- is `github.ref` dynamic to the PR's `HEAD` (or updated for the new commit onto the PR), or would any subsequent action step miss the updated `README.md`? 😦

Reply:
I'm trolling through the earlier steps to see what the actual state is.  The process seems to run as follows:
- We create an empty local Git repo ([log](https://github.com/arcalot/arcaflow-plugin-rtla/actions/runs/11611804740/job/32335255241#step:2:25)):
    ```
    /usr/bin/git init /home/runner/work/arcaflow-plugin-rtla/arcaflow-plugin-rtla
    ```
- then after a bunch of configuration stuff, we fetch the upstream commit that we want to build:
  - for a branch build ([log](https://github.com/arcalot/arcaflow-plugin-rtla/actions/runs/11611804740/job/32335255241#step:2:53)):
    ```
    /usr/bin/git [...] fetch [...] origin +50[...]41d:refs/remotes/origin/initial-plugin
    ```
  - for a PR build ([log](https://github.com/arcalot/arcaflow-plugin-rtla/actions/runs/11611804740/job/32335255241#step:2:53)):
    ```
    /usr/bin/git [...] fetch [...] origin +50[...]41d:refs/remotes/pull/1/merge
    ```
- and then we check out the local ref that we created with `fetch`
  - for a branch ([log](https://github.com/arcalot/arcaflow-plugin-rtla/actions/runs/11612337212/job/32335811050#step:2:62)):
    ```
    /usr/bin/git checkout --progress --force -B initial-plugin refs/remotes/origin/initial-plugin
    ```
  - for the PR ([log](https://github.com/arcalot/arcaflow-plugin-rtla/actions/runs/11611804740/job/32335255241#step:2:62)):
    ```
    /usr/bin/git checkout --progress --force refs/remotes/pull/1/merge
    ```
  Note that, for a branch build but _not_ for a PR build, we create a local branch; for a PR build, we detach the head.

So, for the doc update, I now have the workflow fetching the head ref and checking it out; then it commits the change and pushes it back.  This all seems to be working (given that my PR branch is in the upstream repo...).

Now, I'm trying to restore the checkout to what it was before the doc update, and it's proving to be challenging.  `git switch` seems to be picky, so I've switched back to `git checkout`.  `github.ref` looks _wrong_ relative to what is being used in the initial checkout, which is `refs/remote/pull/<n>/merge` -- it's missing the `remote` in the path!  So, I'm constructing my own path using `github.ref_name`, which is looking promising....

Anyway, to answer your question, `github.ref` refers (more or less...) to a commit which contains the PR's `HEAD` _merged_ on to the target branch -- so it's neither here nor there.  Arguably, I should be updating _both_ the PR branch **and** the merge ref, but I'm not confident that I can change the merge ref, since I see things referring to it by its hash.  And, I suspect that nothing downstream of this point cares whether it's got the updated doc or not:  I think updating the docs is the last thing to run, after building, testing, and exporting the container image.  It's just the `README.md`, so it just needs to be part of the merge, not part of anything before that.

Reply:
> It's just the README.md, so it just needs to be part of the merge, not part of anything before that. So, I think this is fine.

You're probably right, and in a way it's nice to *not* have it as part of the PR review. Then again, if the doc generator does something wacky we don't see it until someone pulls a new PR (or looks on `main`), so there's a tradeoff.

Reply:
And ... OK, I should'a looked first, but there *is* a [docsgen commit on your test PR](https://github.com/arcalot/arcaflow-plugin-rtla/pull/2/commits/a21eb1e9bfc8c9a80f0924babca8d675f0db0494), so I guess that works out... for better or worse ...

Reply:
>  it's nice to not have it as part of the PR review.

Ah, but we don't get off that easy -- the sequence goes like this (assuming that the PR branch is in the upstream repo):
- The author pushes his/her local branch to the upstream repo and posts the PR.
- The CI runs the bot which detects the need for a doc update and adds a commit to the upstream branch.
- The reviewer opens the PR and, lo and behold, the doc update is part of the review.
- The author then has to pull the upstream branch back into his local repo to pick up the change, if s/he has to make any subsequent changes.

So, the bot's changes are subject to supervision, and we have the ability to intercede before they are merged.  (In fact, with this change, updates from the bot won't be able to get in in any other way.)

> there is a [docsgen commit on your test PR](https://github.com/arcalot/arcaflow-plugin-rtla/pull/2/commits/a21eb1e9bfc8c9a80f0924babca8d675f0db0494), so I guess that works out

Right -- that was a key part of the testing:  the workflow had to complete successfully, **_and_** the update had to appear on the branch.  😀